service/history/timer_queue_active_task_executor.go

```diff
@@ -300,12 +300,14 @@ Loop:
 			continue Loop
 		}
 
-		timeoutFailure.GetTimeoutFailureInfo().LastHeartbeatDetails = activityInfo.LastHeartbeatDetails
-		// If retryState is Timeout then it means that expirationTime is expired.
-		// ExpirationTime is expired when ScheduleToClose timeout is expired.
 		if retryState == enumspb.RETRY_STATE_TIMEOUT {
-			timeoutFailure.GetTimeoutFailureInfo().TimeoutType = enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE
+			// If retryState is Timeout then it means that expirationTime is expired.
+			// ExpirationTime is expired when ScheduleToClose timeout is expired.
+			timeoutType := enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE
+			failureMsg := fmt.Sprintf("activity %v timeout", timeoutType.String())
```

Comment:
```suggestion
			const timeoutType = enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE
			const failureMsg = fmt.Sprintf("activity %v timeout", timeoutType.String())
```
? I think compiler should inline consts then.

Reply:
`failureMsg` can't be const. I changed to timeoutType to be const.